pkg/terraform/gather/ovirt/ip.go

```diff
@@ -2,13 +2,126 @@
 package ovirt
 
 import (
+	//"context"
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/openshift/installer/pkg/asset/installconfig/ovirt"
 	"github.com/openshift/installer/pkg/terraform"
+	ovirtsdk4 "github.com/ovirt/go-ovirt"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	//"time"
 )
 
+func checkPortIsOpen(host string, port string) bool {
+	timeout := time.Second
+	conn, err := net.DialTimeout("tcp", net.JoinHostPort(host, port), timeout)
+	if err != nil {
+		logrus.Debugln(fmt.Sprintf("connection error: %v", err))
+	}
+	if conn != nil {
+		defer conn.Close()
+		return true
+	}
+	return false
+}
+
+func getreportedDevices(c *ovirtsdk4.Connection, vmID string) (*ovirtsdk4.ReportedDeviceSlice, error) {
+
+	vmsService := c.SystemService().VmsService()
+	//# Look up fot the vm by id:
+	vmResp, err := vmsService.VmService(vmID).Get().Send()
+	if err != nil {
+		return nil, fmt.Errorf("failed to find VM, by id %v, reason: %v", vmID, err)
+	}
+	vm := vmResp.MustVm()
+
+	// Get the reported-devices service for this vm:
+	reportedDevicesService := vmsService.VmService(vm.MustId()).ReportedDevicesService()
+
+	// Get the guest reported devices
+	reportedDeviceResp, err := reportedDevicesService.List().Send()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get reported devices list, reason: %v", err)
+	}
+	reportedDeviceSlice, _ := reportedDeviceResp.ReportedDevice()
+
+	if len(reportedDeviceSlice.Slice()) == 0 {
+		return nil, fmt.Errorf("cannot find IPs for vmId: %s", vmID)
+	}
+
+	return reportedDeviceSlice, nil
+
+}
+
+func findVirtualMachineIP(c *ovirtsdk4.Connection, moRefValue string) (string, error) {
+
+	ipAddress := ""
+
+	reportedDeviceSlice, err := getreportedDevices(c, moRefValue)
+	if err != nil {
+		return "", err
+	}
+
+	for _, reportedDevice := range reportedDeviceSlice.Slice() {
+		ips := reportedDevice.MustIps()
+
+		if ips != nil {
+			for _, ip := range ips.Slice() {
+				ipres := ip.MustAddress()
+				if ipres != "" {
+					if checkPortIsOpen(ipres, "22") {
+						ipAddress = ipres
+					}
+				}
+
+			}
+		}
+	}
+
+	if ipAddress == "" {
+		return "", fmt.Errorf("coudlnt find usable bootstrap IP address for vm id: %s", moRefValue)
+	}
+
+	logrus.Debugln(fmt.Sprintf("ovirt vm id: %s , found usable bootstrap IP %s", moRefValue, ipAddress))
+	return ipAddress, nil
+}
+
 // BootstrapIP returns the ip address for bootstrap host.
 // still unsupported, because qemu-ga is not available - see https://bugzilla.redhat.com/show_bug.cgi?id=1764804
 func BootstrapIP(tfs *terraform.State) (string, error) {
-	return "", nil
+
+	client, err := ovirt.NewConnection()
+	if err != nil {
+		return "", fmt.Errorf("failed to initialize connection to ovirt-engine's %s", err)
+	}
+	defer client.Close()
+
+	br, err := terraform.LookupResource(tfs, "module.bootstrap", "ovirt_vm", "bootstrap")
```

Comment:
Why not look for the bootstrap VM in ovirt without terraform? less deps == less problems :)

You can use the same logic as https://github.com/openshift/cluster-api-provider-ovirt/blob/master/pkg/cloud/ovirt/clients/machineservice.go#L275-L291

Reply:
I think since we operate here externally to the cluster using the id is more reliable, plus it's similar to what other vendors are doing [1].

[1] https://github.com/openshift/installer/blob/644a93e5ab28912d0a19171883d489ac47b3faea/pkg/terraform/gather/vsphere/ip.go#L58

Reply:
+1